core/tee/tee_rpmb_fs.c

```diff
@@ -1105,12 +1108,18 @@ static TEE_Result tee_rpmb_init(uint16_t dev_id)
 		DMSG("RPMB INIT: Verifying Key");
 
 		res = tee_rpmb_verify_key_sync_counter(dev_id);
-		if (res != TEE_SUCCESS && !rpmb_ctx->key_verified) {
+		if (res == TEE_ERROR_ITEM_NOT_FOUND &&
+			!rpmb_ctx->key_verified) {
 			/*
 			 * Need to write the key here and verify it.
 			 */
+			DMSG("RPMB INIT: Auth key not yet written");
 			res = tee_rpmb_write_and_verify_key(dev_id);
 		}
+		else if (res != TEE_SUCCESS) {
```

Comment:
Should this be in an `else` maybe a plain `if` is more suitable? Or miving the check/print into the corresponding `if` instead?

Reply:
Sorry but I don't get how an `if` is more suitable or where the corresponding `if` is.
The `else` is a catch-all for all errors that is not `TEE_ERROR_ITEM_NOT_FOUND`.
`TEE_ERROR_ITEM_NOT_FOUND` is 'mapped' to `RPMB_RESULT_AUTH_KEY_NOT_PROGRAMMED`. The RPMB device will keep returning this error until a key is programmed/written, so if this is the error returned, we want to write (and verify) a key. Else for all other errors, i.e. key already written, but verification failed, we print an error message stating so.